### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by dnKaratzas**
- Moved logic's to the async methods
- Removed continuations
- The closure methods now call the async methods
- Removed OpenCombine dependency

### CODE REVIEW

**Comment by DarthMike** [comment]
Will take a while to review; lots of changes,  many just formatting making it hard to check all code is fine.

**Comment by dnKaratzas** [comment]
> Will take a while to review; lots of changes, many just formatting making it hard to check all code is fine.

Need to make the #247  to avoid formatting issues. Will start working on it

**Review by DarthMike — COMMENTED** [review]
Only 1 comment, everything else looks good

File: web3swift/src/Client/BaseEthereumClient.swift [hunk 1]
```diff
@@ -26,105 +25,97 @@ public class BaseEthereumClient: EthereumClientProtocol {
...
-                if let resString = data as? String {
-                    let network = EthereumNetwork.fromString(resString)
-                    completionHandler(.success(network))
-                } else {
-                    completionHandler(.failure(.unexpectedReturnValue))
-                }
-            case .failure(let error):
-                self.failureHandler(error, completionHandler: completionHandler)
+        do {
+            let data = try await networkProvider.send(method: "net_version", params: [emptyParams], receive: String.self)
```
[the remark this answers is not in the document]
Comment (dnKaratzas) [reply web3swift/src/Client/BaseEthereumClient.swift]: Done!

**Review by DarthMike — APPROVED** [review]

**Pull request merged**